conjure-undertow-processor/src/main/java/com/palantir/conjure/java/undertow/processor/data/ParamTypesResolver.java

```diff
@@ -349,6 +354,32 @@ private CodeBlock getCollectionParamDecoder(
                 });
     }
 
+    private Optional<CodeBlock> getParamDecoderConstructor(DeclaredType declaredType) {
+        TypeElement typeElement = (TypeElement) declaredType.asElement();
+        // new T(String)
+        return typeElement.getEnclosedElements().stream()
+                .filter(element -> element.getKind() == ElementKind.CONSTRUCTOR)
+                .map(ExecutableElement.class::cast)
+                .filter(element -> element.getThrownTypes().stream().allMatch(this::isRuntimeException)
+                        && !element.getParameters().isEmpty())
+                .map(element -> {
+                    List<CodeBlock> args = new ArrayList<>();
+                    for (VariableElement parameter : element.getParameters()) {
+                        TypeName parameterTypeName = TypeName.get(parameter.asType());
+                        if (parameterTypeName.equals(ClassName.get(UndertowRuntime.class))) {
+                            args.add(CodeBlock.of("runtime"));
+                        } else if (parameterTypeName.equals(ClassName.get(Endpoint.class))) {
+                            args.add(CodeBlock.of("this"));
+                        } else {
+                            return null;
+                        }
+                    }
+                    return CodeBlock.of("new $T($L)", TypeName.get(declaredType), CodeBlock.join(args, ", "));
+                })
+                .filter(Objects::nonNull)
+                .findFirst();
```

Comment:
I opted to just select the first matching constructor that we find, rather than try to prioritize them in some way. This is consistent with the behavior of similar methods here and almost certainly will be acceptable in real-world use.

Reply:
We could produce the (first?) constructor with `max(num-parameters)` of the available options for more obvious behavior -- or possibly better still, require exactly zero or one matches, failing compilation otherwise (so it's obvious that the ctor choice is ambiguous)

Reply:
I've also replaced `.findFirst()` with `.collect(MoreCollectors.toOptional()` in the methods used in `getUnknownDecoderFactoryFunction`. Those filters are specific enough where we should never have more than one element, but always better to enforce the expectation rather than assume it.